src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -3024,17 +3111,20 @@ void ONNXSliceOp::tryFold() {
     auto valueAttrSize = valueAttribute.size();
     if (start < 0) {
       start = start + valueAttrSize;
+      if (start < 0)
```

Comment:
I am not 100% sure about the case that is covered here. Numpy documentation doesn't mention anything about negative index > size of array. (https://numpy.org/doc/stable/reference/arrays.indexing.html)

Reply:
I don't think it is necessary to have that check for `start`.  Although for `end` it is necessary. Based on documentation it is valid to use `INT_MAX` (with positive step) and `INT_MIN` (with negative step). The check for `start` is only for safety.